src/ngx_http_lua_ssl.c

```diff
@@ -34,4 +38,169 @@ ngx_http_lua_ssl_init(ngx_log_t *log)
 }
 
 
+static size_t
+ngx_http_lua_ssl_get_error(u_long e, u_char *ssl_err,
+    size_t ssl_err_len, const char *default_err, size_t default_err_len)
+{
+    size_t             len;
+
+    if (e == 0) {
+        len = ngx_min(ssl_err_len, default_err_len);
+        ngx_memcpy(ssl_err, default_err, len);
+
+        return len;
+    }
+
+    ERR_error_string_n(e, (char *) ssl_err, ssl_err_len);
+
+    return ngx_strlen(ssl_err);
+}
+
+
+#ifndef NGX_LUA_NO_FFI_API
+
+
+void *
+ngx_http_lua_ffi_ssl_ctx_init(ngx_uint_t protocols, char **err)
+{
+    ngx_ssl_t           ssl;
+
+    ssl.log = ngx_cycle->log;
+    if (ngx_ssl_create(&ssl, protocols, NULL) != NGX_OK) {
```

Comment:
Well. inherit existing configurations from `ngx_http_lua_loc_conf_t`s ssl field will required this API need `ngx_http_request_t`. we should allow it run any phase. Especially, it can be allowed to run in `init_by_lua_*` to easily cache it. May we expose these FFI  API (like ciphers, crl, ca and so on) to Lua Land . It will be more  flexible and effective to feed arguments to SSL_CTX object.